src/renderers/shaders/ShaderChunk/bsdfs.glsl

```diff
@@ -0,0 +1,152 @@
+//#define ENERGY_PRESERVING_MONOCHROME
+
+float calcLightAttenuation( float lightDistance, float cutoffDistance, float decayExponent ) {
+
+	if ( decayExponent > 0.0 ) {
+
+	  return pow( saturate( -lightDistance / cutoffDistance + 1.0 ), decayExponent );
+
+	}
+
+	return 1.0;
+
+}
+
+
+void BRDF_Lambert( const in IncidentLight incidentLight, const in GeometricContext geometryContext, const in vec3 diffuseColor, inout ReflectedLight reflectedLight ) {
+
+	float lambertianReflectance = saturate( dot( geometryContext.normal, incidentLight.direction ) );
```

Comment:
Did you know that the N * L is actually "Lambert's cosine law" which governs diffuse reflection?  https://en.wikipedia.org/wiki/Lambert%27s_cosine_law

If you read the wikipedia article entitled "Lambertian reflectance", the third sentence states that it follows Lambert's cosine law:

> More technically, the surface's luminance is isotropic, and the luminous intensity obeys Lambert's cosine law.

It is part of the generalized Rendering Equation as well, so you are not wrong that it is part of the more general rendering equation, but it is also Lambertian Reflectance.